src/DeploymentsManager.ts

```diff
@@ -1477,6 +1477,28 @@ export class DeploymentsManager {
     return this.getDeploymentNetworkName();
   }
 
+  private async impersonateAccounts(unknownAccounts: string[]) {
+    if (
+      !this.impersonateUnknownAccounts ||
+      process.env.HARDHAT_DEPLOY_NO_IMPERSONATION
+    ) {
+      return;
+    }
+
+    const isHardhat = this.network.name === 'hardhat';
+    if (isHardhat || this.network.autoImpersonate) {
```

Comment:
since the hardhat network config as autoImpersonate, we need to cater for the case, the use explicity set it to to `autoImpersonate: false`

so I would remove isHardhat here and instead set that up in the loading of the network config

so here we can just do if (this.network.autoImpersinate)

Reply:
👍 

Please check out the last update

Reply:
Hey @wighawag,
Would it be possible to merge this? It'd be great to use the official version in our repo instead of my fork 😅